### PR TITLE
spanner: Expand test coverage for ResultSets

### PR DESCRIPTION
**Pull request by nithinsujir**
ResultSets coverage now at 99%.

### CODE REVIEW

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java [hunk 1]
```diff
@@ -75,12 +179,52 @@ public void resultSetIteration() {
...
     assertThat(rs.next()).isTrue();
     assertThat(rs.getCurrentRowAsStruct()).isEqualTo(struct2);
     assertThat(rs.getString(0)).isEqualTo("y");
     assertThat(rs.getLong(1)).isEqualTo(3L);
     assertThat(rs.isNull(2)).isTrue();
     assertThat(rs.next()).isFalse();
+
+    try {
+      rs.getStats();
+      fail("Exception expected");
```
Comment (snehashah16) [inline google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: this will fail after my change for DML .. would u mind holding off on ur CL just so that I dont have to redo my PR for DML/PDML.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java [hunk 2]
```diff
@@ -34,11 +41,51 @@
 
   @Test
   public void resultSetIteration() {
+    double dval = 1.2;
+    String s = "s";
+    String byt = "101";
```
Comment (snehashah16) [inline google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: is this testing string or byte ?

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java [hunk 3]
```diff
@@ -34,11 +41,51 @@
 
   @Test
   public void resultSetIteration() {
```
Comment (snehashah16) [inline google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: its easier to read test case when they are as explicit as possible. ie - variable names could be more informative (even if long)

also testing edge cases or large values provide surface errors than smaller value.. e.g. - testing a large random bytes value

Comment (nithinsujir) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: I don't disagree if we're testing the implementation, but the purpose of this change is to expand the line coverage, i.e to ensure execution of paths. For edge case testing etc., we would need to add a lot more specific cases for each data type.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java (hunk 2 above)
Comment (nithinsujir) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: byte.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java (hunk 1 above)
Comment (nithinsujir) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: Sure, not a problem. Can you elaborate for my knowledge why it will fail?

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java (hunk 3 above)
Comment (snehashah16) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: i was hoping to shoot 2 birds with 1 stone, though i don't mean or need to have extensive edge case testing with this exercise either.

File: google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java (hunk 1 above)
Comment (snehashah16) [reply google-cloud-clients/google-cloud-spanner/src/test/java/com/google/cloud/spanner/ResultSetsTest.java]: rs.getStats() returns null instead of an exception ..

**Comment by nithinsujir** [comment]
Marking this WIP until DML changes are in.

**Comment by nithinsujir** [comment]
@pongad Merged DML changes. Turns out there is no impact because the getStats() that DML changed is in a different class than the one in this test.

This PR is ready to go.

**Comment by pongad** [comment]
We're transitioning ownership of this repo and I don't have write access anymore. @chingor13 @JesseLovelace can you help?

**Comment by chingor13** [comment]
Does this need to go out with the spanner DML feature or can it wait until after the release is completed?

**Comment by nithinsujir** [comment]
It can wait until after the release. It's unrelated to DML.

**Comment by chingor13** [comment]
SGTM
